metadata-ingestion/src/datahub/ingestion/source/file.py

```diff
@@ -205,17 +205,17 @@ def get_filenames(self) -> Iterable[str]:
             self.report.total_num_files = 1
             return [str(self.config.path)]
 
-    def get_workunits(self) -> Iterable[Union[MetadataWorkUnit, UsageStatsWorkUnit]]:
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
         return auto_workunit_reporter(self.report, self.get_workunits_internal())
 
     def get_workunits_internal(
         self,
-    ) -> Iterable[Union[MetadataWorkUnit, UsageStatsWorkUnit]]:
+    ) -> Iterable[MetadataWorkUnit]:
         for f in self.get_filenames():
             for i, obj in self.iterate_generic_file(f):
                 id = f"file://{f}:{i}"
                 if isinstance(obj, UsageAggregationClass):
-                    yield UsageStatsWorkUnit(id, obj)
+                    logger.warning(f"Dropping deprecated UsageAggregationClass: {obj}")
```

Comment:
let's use self.report.report_warning here

Reply:
I used to have the logic here, but this actually won't ever get called (I need to remove this case) because I now filter these when reading the file, to fix type issues. How necessary do you think this is?